src/Morphic-Widgets-Basic/PluggableButtonMorph.class.st

```diff
@@ -497,17 +479,16 @@ PluggableButtonMorph >> enabled: aBoolean [
 ]
 
 { #category : #accessing }
-PluggableButtonMorph >> extent: aPoint [ 
-	"Set the receiver's extent to value provided.
-	Update the gradient fills."
+PluggableButtonMorph >> extent: aPoint [
+	"Set the receiver's extent to value provided."
 
-	|answer|
-	aPoint = self extent ifTrue: [^super extent: aPoint].
+	| answer |
+	aPoint = self extent ifTrue: [ ^ super extent: aPoint ].
+	
 	answer := super extent: aPoint.
-	self gradientLook ifTrue: [
-		self fillStyle isOrientedFill
-			ifTrue: [self fillStyle: self fillStyleToUse]].
-	^answer
+	
+	self fillStyle isOrientedFill ifTrue: [ self fillStyle: self fillStyleToUse ].
```

Comment:
remove this line as well. (the whole method can be removed and just rely on super

Reply:
If a theme use a full style with a gradient, will it be updated without this line?

Reply:
it depents on the gradient and how it is updated wrt the button, but most fillstyles support dynamic sizing and those who dont could easily be modified / extended if this is what one wants. It should not be the responsibility of the button to take care of this edge case

Reply:
Ok I see now and I agree with you! :)